standards/core/peer-exchange.md

```diff
@@ -66,26 +66,45 @@ message PeerInfo {
   bytes enr = 1;
 }
 
-message PeerExchangeQuery {
+message PeerExchangeRequest {
   uint64 num_peers = 1;
 }
 
 message PeerExchangeResponse {
   repeated PeerInfo peer_infos = 1;
 }
 
+message PeerExchangeResponseStatus {
+  uint32 status = 1;
```

Comment:
why not add these fields to existing PeerExchangeResponse itself? That way it makes it backward compatible.

Reply:
> why not add these fields to existing PeerExchangeResponse itself? That way it makes it backward compatible.

What do you mean? Isn't that the opposite? That would break the protocol. With this solution I tried to avoid breaking the prev protocol by adding it separately under a new index.

Reply:
afaiu adding new fields to existing messages doesn't break existing protocol [ref](https://stackoverflow.com/questions/45854645/google-protobuf-if-i-add-field-from-one-side-will-it-break-another-one), in fact that is the way protobufs are extendible in nature. New fields are ignored and not decoded by an older version.

Reply:
Ok, I think we are on the same page. Thank you for the research btw. I also had this in mind.
So I did not embed the status into the response to keep backward compatibility.
But as I went through the docs, which does not clearly indicate that extending an embedded type still maintains compatibility, now I read another sources whereas with protobuf3 all fields are like "optional" and will have a default (according to the type). So backward and forward compatibility is kept.

If that is acceptable I will change this as suggested (also in nwaku code) that rather have the status in the response field.
Also will remove the optional from the fields and leave to the callee/caller side to decide which they read.
Minimal checks can still be done through the status field.

Do you agree?